Always shift solar influx by 30 minutes

zoltanmaric opened the pull request:
<!--
SPDX-FileCopyrightText: 2021 The Atlite Authors

SPDX-License-Identifier: CC0-1.0
-->

## Always Shift Solar Influx by 30 Minutes

<!--- Provide a general, short summary of your changes in the title above -->

## Description
Fixes a bug in solar position time shift for ERA5 influx data where for some cases the frequency of the data would be incorrectly inferred. The change always shifts by -30 minutes, regardless of the sampling of the data. See discussion in https://github.com/PyPSA/atlite/issues/256#issuecomment-1268473262 for reasoning.
<!--- Describe your changes in detail -->

## Motivation and Context
Closes #256

## How Has This Been Tested?
* The [first commit](c190c3410b037e278a632322dff400311e925bb8) in the pull request introduces a [failing unit test](https://github.com/zoltanmaric/atlite/blob/90c593142371d9a2a9c97e5a13d5d128594a84cc/test/test_preparation_and_conversion.py#L501-L504) that demonstrates the bug reported in #256.
* The [second commit](90c593142371d9a2a9c97e5a13d5d128594a84cc) fixes the bug by always shifting by 30 minutes (removing the inference of the frequency of the data)
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I tested my contribution locally and it seems to work fine.
- [x] I locally ran `pytest` inside the repository and no unexpected problems came up.
- [x] I have adjusted the docstrings in the code appropriately.
- [ ] I have documented the effects of my code changes in the documentation `doc/`.
- [ ] I have added newly introduced dependencies to `environment.yaml` file.
- [x] I have added a note to release notes `doc/release_notes.rst`.
- [x] I have used `pre-commit run --all` to lint/format/check my contribution

FabianHofmann reviewed (commented):
@zoltanmaric this looks great! Thank you for raising **and** tackling this issue. One question: What would happen if the time resolution wasn't 1 hour but, let's say, 3 hours? This would still work right?

zoltanmaric commented:
> @zoltanmaric this looks great! Thank you for raising **and** tackling this issue. One question: What would happen if the time resolution wasn't 1 hour but, let's say, 3 hours? This would still work right?

Yes, it works at any time sampling. I can add a unit test for that to this PR if you like.

That said, there does seem to be a slight difference between 1-hour sampling and all other samplings (I tried 2h, 3h, 4h, and 6h) - see my comment here: https://github.com/PyPSA/atlite/issues/256#issuecomment-1271384837

zoltanmaric commented on test/test_preparation_and_conversion.py:
```diff
@@ -293,12 +298,6 @@ def coefficient_of_performance_test(cutout):
-SARAH_DIR = os.getenv("SARAH_DIR", "/home/vres/climate-data/sarah_v2")
-GEBCO_PATH = os.getenv("GEBCO_PATH", "/home/vres/climate-data/GEBCO_2014_2D.nc")
-
-
```
This section was just moved to the top of the file, because I added a reference to `TIME` in one of the methods above.

zoltanmaric commented:
@FabianHofmann I added a test for 3h-sampling in 4de756a. Let me know if you want to keep it or if I should remove it.

FabianHofmann commented:
> @FabianHofmann I added a test for 3h-sampling in [zoltanmaric@17a0847](https://github.com/zoltanmaric/atlite/commit/17a08475b4c3d97d3a380ea8a2d4b9fb3e6c3b2c). Let me know if you want to keep it or if I should remove it.

Let's keep the test. As far as I see, only a release note is missing. Would you be so kind? Then we can merge it.

zoltanmaric commented:
> As far as I see, only a release note is missing. Would you be so kind? Then we can merge it.

I wasn't sure about the format - I added a new version heading (`0.2.10`) and added a note there (see [here](https://github.com/PyPSA/atlite/compare/master...zoltanmaric:atlite:time-shift-30min#diff-9b07fd28838e759e05bdaa5293886d860374fe291bdc6cc43c20f951fc77eab4R14-R19)). Is that right?

FabianHofmann commented on RELEASE_NOTES.rst:
```diff
@@ -11,6 +11,13 @@ Release Notes
 .. Upcoming Release
 .. =================
 
+Version 0.2.10
+=================
```
```suggestion
Version 0.2.10
==============
```

FabianHofmann replied:
(otherwise there will be a warning during the compilation)

zoltanmaric replied:
OK, I also noticed in the meantime that the latest release is `0.2.8`. Should I then add my note to version `0.2.9` instead?

FabianHofmann replied:
No, there was just the GH release missing (which is sort of secondary). But perhaps you can name the heading `Upcoming Release`?

FabianHofmann replied:
See suggestion.

zoltanmaric replied:
Done!
